src/machines/java/com/enderio/machines/common/block/MachineBlock.java

```diff
@@ -80,20 +85,21 @@ private void updateBlockEntityCache(LevelReader level, BlockPos pos) {
     }
 
     @Override
-    public InteractionResult use(BlockState pState, Level pLevel, BlockPos pPos, Player pPlayer, InteractionHand pHand, BlockHitResult pHit) {
-        if (pPlayer.getItemInHand(pHand).is(EIOItems.YETA_WRENCH.get())) {
-            return InteractionResult.PASS;
-        }
-
-        if (pLevel.isClientSide) {
+    public InteractionResult use(BlockState state, Level level, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit) {
+        ItemStack heldItem = player.getItemInHand(hand);
+        if (level.isClientSide() || hand != InteractionHand.MAIN_HAND){
             return InteractionResult.SUCCESS;
-        } else {
-            MenuProvider menuprovider = this.getMenuProvider(pState, pLevel, pPos);
-            if (menuprovider != null && pPlayer instanceof ServerPlayer serverPlayer) {
-                NetworkHooks.openScreen(serverPlayer, menuprovider, buf -> buf.writeBlockPos(pPos));
+        }
+        BlockEntity entity = level.getBlockEntity(pos);
+        //pass on the use command to corresponding block entity.
+        InteractionResult result = ((MachineBlockEntity)entity).onBlockEntityUsed(state, level, pos, player, hand,hit);
```

Comment:
While I don't dislike the change to pass it on, do you already have a machine in mind for this?

Reply:
Yes. I do. PR coming very soon in a repository near you. I'm currently rebuilding the fluid interface, and clicking on a fluid containing machine with a bucket should have the effect that the bucket should fill up. This makes it extremely convenient to do with this method instead of making a separate class for every machine block containing fluid and then binding a separate method to pass the event. You just hook into this event in the already existing entity class (Which I have used for the vat for example, but again coming soon hopefully).